src/mlpack/core/data/load_numeric.hpp

```diff
@@ -13,117 +13,156 @@
 #ifndef MLPACK_CORE_DATA_LOAD_NUMERIC_HPP
 #define MLPACK_CORE_DATA_LOAD_NUMERIC_HPP
 
+#include "text_options.hpp"
+#include "load_categorical.hpp"
+#include "load_dense.hpp"
+#include "load_sparse.hpp"
+
 namespace mlpack {
 namespace data {
 
-// These help assemble the Armadillo csv_opts flags.
-inline const arma::csv_opts::opts& NoTransposeOpt(const bool noTranspose)
-{
-  if (noTranspose)
-    return arma::csv_opts::none;
-  else
-    return arma::csv_opts::trans;
-}
-
-inline const arma::csv_opts::opts& HasHeadersOpt(const bool hasHeaders)
-{
-  if (hasHeaders)
-    return arma::csv_opts::with_header;
-  else
-    return arma::csv_opts::no_header;
-}
-
-inline const arma::csv_opts::opts& SemicolonOpt(const bool semicolon)
+template<typename MatType>
+bool LoadNumeric(const std::string& filename,
+                 MatType& matrix,
+                 std::fstream& stream,
+                 TextOptions& opts)
 {
-  if (semicolon)
-    return arma::csv_opts::semicolon;
-  else
-    return arma::csv_opts::none;
-}
+  bool success = false;
 
-inline const arma::csv_opts::opts& MissingToNanOpt(const bool missingToNan)
-{
-  if (missingToNan)
+  TextOptions txtOpts(std::move(opts));
+  if constexpr (IsSparseMat<MatType>::value)
+  {
+    success = LoadSparse(filename, matrix, txtOpts, stream);
+  }
+  else if (txtOpts.Categorical() ||
+      (txtOpts.Format() == FileType::ARFFASCII))
+  {
+    success = LoadCategorical(filename, matrix, txtOpts);
+  }
+  else if constexpr (IsCol<MatType>::value)
   {
-    #if ARMA_VERSION_MAJOR >= 12
-    return arma::csv_opts::strict;
-    #else
-    return arma::csv_opts::none;
-    #endif
+    success = LoadCol(filename, matrix, txtOpts, stream);
+  }
+  else if constexpr (IsRow<MatType>::value)
+  {
+    success = LoadRow(filename, matrix, txtOpts, stream);
```

Comment:
Didn't many of these get moved to `load_deprecated.hpp`?  Maybe it makes sense to move them into `load_dense.hpp` and call them e.g. `LoadDenseRow()` and `LoadDenseCol()` instead?